inbox/events/microsoft/parse.py

```diff
@@ -81,6 +81,14 @@ def parse_msgraph_datetime_tz_as_utc(datetime_tz: MsGraphDateTimeTimeZone):
     # so we need to use ciso8601 because Python stdlib only supports up to 6.
     dt = ciso8601.parse_datetime(datetime_tz["dateTime"])
 
+    # Microsoft sometimes returns payloads such as this one:
+    # {"dateTime": "0001-01-01T00:00:00.0000000Z", "timeZone": "tzone://Microsoft/Utc"}
+    # We're removing the timezone from the parsed datetime, otherwise
+    # `tzinfo.localize` would fail. This datetime is obviously not correct
+    # anyway.
+    if dt.tzinfo and dt == datetime.datetime(1, 1, 1, tzinfo=pytz.UTC):
```

Comment:
Nit, I think it would be enough to
```suggestion
    if dt == datetime.datetime(1, 1, 1, tzinfo=pytz.UTC):
```

Reply:
Did it for performance reasons. I'm generally against premature optimisation, but I don't know how often this is called, so I did not want to decrease performance for all executions here, and a single `dt.tzinfo` check doesn't hurt.

By the way, it's `1000ms` vs `70ms` for 1,000,000 calls.

Reply:
For both minor performance gains and more importantly for readability, I suggest we construct `datetime.datetime(1, 1, 1, tzinfo=pytz.UTC)` only once and assign it to a const like `MICROSOFT_NO_DATETIME_SENTINEL_VALUE`, along with a explaining what this is about.

Reply:
Even if we do this, I'd still like to keep the `dt.tzinfo` check as it's extremely cheap compared to datetime comparison (which could be a bit more expensive when comparing two datetimes using different timezone objects for UTC, e.g. `pytz.UTC` and `datetime.timezone.utc`)

Reply:
Yep makes sense